openforcefield/tests/test_forcefield.py

```diff
@@ -1112,6 +1179,161 @@ def test_freesolv_parameters_assignment(self, freesolv_id, forcefield_version, a
+                     'OBC1': xml_gbsa_ff_obc1,
+                     'OBC2': xml_gbsa_ff_obc2,
+                     }
+        # Create OpenFF System with the current toolkit.
+        ff = ForceField('test_forcefields/smirnoff99Frosst.offxml',
+                        off_gbsas[gbsa_model])
+        off_top = molecule.to_topology()
+        if is_periodic:
+            off_top.box_vectors = ((30., 0, 0), (0, 30., 0), (0, 0, 30.)) * unit.angstrom
+
+        else:
+            off_top.box_vectors = None
+
+        off_omm_system = ff.create_openmm_system(off_top, charge_from_molecules=[molecule])
+
+        off_nonbonded_force = [force for force in off_omm_system.getForces() if
+                               isinstance(force, openmm.NonbondedForce)][0]
+
+
+        omm_top = off_top.to_openmm()
+        pmd_struct = pmd.openmm.load_topology(omm_top, off_omm_system, positions)
+        prmtop_file = NamedTemporaryFile(suffix='.prmtop')
+        inpcrd_file = NamedTemporaryFile(suffix='.inpcrd')
+        pmd_struct.save(prmtop_file.name, overwrite=True)
+        pmd_struct.save(inpcrd_file.name, overwrite=True)
+
+        openmm_gbsas = {'HCT': openmm.app.HCT,
+                        'OBC1': openmm.app.OBC1,
+                        'OBC2': openmm.app.OBC2,
+                        }
+
+        # The functional form of the nonbonded force will change depending on whether the cutoff
+        # is None during initialization. Therefore, we need to figure that out here.
+
+        # WARNING: The NonbondedMethod enums at openmm.app.forcefield and openmm.CustomGBForce
+        # aren't necessarily the same, and could be misinterpreted if the wrong one is used. For
+        # create_system_from_amber, we must provide the app.forcefield version.
+
+        if is_periodic:
+            amber_nb_method = openmm.app.forcefield.CutoffPeriodic
+            amber_cutoff = off_nonbonded_force.getCutoffDistance()
+        else:
+            amber_nb_method = openmm.app.forcefield.NoCutoff
+            amber_cutoff = None
+
+        (amber_omm_system,
+         amber_omm_topology,
+         amber_positions) = create_system_from_amber(prmtop_file.name,
+                                                     inpcrd_file.name,
+                                                     implicitSolvent=openmm_gbsas[gbsa_model],
+                                                     nonbondedMethod = amber_nb_method,
+                                                     nonbondedCutoff=amber_cutoff,
+                                                     gbsaModel='ACE',
+                                                     implicitSolventKappa=0.,
+                                                     )
+
+        off_gbsa_force = [force for force in off_omm_system.getForces() if
+                              (isinstance(force, openmm.GBSAOBCForce) or
+                               isinstance(force, openmm.openmm.CustomGBForce))][0]
```

Comment:
As a paranoid check, I'd do a `assert len(off_gbsa_forces) == 1` before retrieving the first element.

Reply:
Good idea! Check added in 7ae17c1